pkg/authorization/authorizer/attributes.go

```diff
@@ -34,6 +35,16 @@ func (a DefaultAuthorizationAttributes) RuleMatches(rule authorizationapi.Policy
 
 		if a.resourceMatches(allowedResourceTypes) {
 			if a.nameMatches(rule.ResourceNames) {
+				// this rule matches the request, so we should check the additional restrictions to be sure that it's allowed
+				if rule.AttributeRestrictions.Object != nil {
+					switch rule.AttributeRestrictions.Object.(type) {
+					case (*authorizationapi.AllowPersonalSubjectAccessReview):
```

Comment:
I think `IsPersonalSubjectAccessReview` is a better name for the restriction marker type itself